pkg/kubefedctl/unjoin.go

```diff
@@ -188,86 +189,88 @@ func UnjoinCluster(hostConfig, clusterConfig *rest.Config, federationNamespace,
 		return err
 	}
 
-	var deletionSucceeded bool
 	if clusterClientset != nil {
-		deletionSucceeded = deleteRBACResources(clusterClientset, federationNamespace, unjoiningClusterName, hostClusterName, dryRun)
+		err := deleteRBACResources(clusterClientset, federationNamespace, unjoiningClusterName, hostClusterName, forceDeletion, dryRun)
+		if err != nil && !forceDeletion {
+			return err
+		}
 
 		err = deleteFedNSFromUnjoinCluster(hostClientset, clusterClientset, federationNamespace, unjoiningClusterName, dryRun)
-		if err != nil {
-			klog.Errorf("Error deleting federation namespace from unjoin cluster: %v", err)
-			deletionSucceeded = false
+		if err != nil && !forceDeletion {
+			return err
 		}
 	}
 
 	// deletionSucceeded when all operations in deleteRBACResources and deleteFedNSFromUnjoinCluster succeed.
-	if deletionSucceeded || forceDeletion {
-		deleteFederatedClusterAndSecret(hostClientset, client, federationNamespace, unjoiningClusterName, dryRun)
-	}
-
-	return nil
+	return deleteFederatedClusterAndSecret(hostClientset, client, federationNamespace, unjoiningClusterName, forceDeletion, dryRun)
 }
 
 // deleteFederatedClusterAndSecret deletes a federated cluster resource that associates
 // the cluster and secret.
 func deleteFederatedClusterAndSecret(hostClientset kubeclient.Interface, client genericclient.Client,
-	federationNamespace, unjoiningClusterName string, dryRun bool) {
+	federationNamespace, unjoiningClusterName string, forceDeletion, dryRun bool) error {
 	if dryRun {
-		return
+		return nil
 	}
 
-	klog.V(2).Infof("Deleting federated cluster resource from namespace: %s for unjoin cluster: %s",
+	klog.V(2).Infof("Deleting federated cluster resource from namespace %q for unjoin cluster %q",
 		federationNamespace, unjoiningClusterName)
 
 	fedCluster := &fedv1a1.FederatedCluster{}
 	err := client.Get(context.TODO(), fedCluster, federationNamespace, unjoiningClusterName)
 	if err != nil {
-		klog.Errorf("Failed to get FederatedCluster resource from namespace: %s for unjoin cluster: %s due to: %v", federationNamespace, unjoiningClusterName, err)
-		return
+		return errors.Wrapf(err, "Failed to get federated cluster \"%s/%s\"", federationNamespace, unjoiningClusterName)
 	}
 
 	err = hostClientset.CoreV1().Secrets(federationNamespace).Delete(fedCluster.Spec.SecretRef.Name,
 		&metav1.DeleteOptions{})
-	if err != nil {
-		klog.Errorf("Failed to delete Secret resource from namespace: %s for unjoin cluster: %s due to: %v", federationNamespace, unjoiningClusterName, err)
+	if apierrors.IsNotFound(err) {
+		klog.V(2).Infof("Secret \"%s/%s\" does not exist in the host cluster.", federationNamespace, fedCluster.Spec.SecretRef.Name)
+	} else if err != nil {
+		message := fmt.Sprintf("Failed to delete secret \"%s/%s\" for unjoin cluster %q",
+			federationNamespace, fedCluster.Spec.SecretRef.Name, unjoiningClusterName)
+		if !forceDeletion {
+			return errors.Wrap(err, message)
+		}
+		klog.V(2).Infof("%s.", message)
```

Comment:
The wrapped error should be logged, not just the message.  Also, why is the period not included directly in the message?  Same comment on: 

 - [L248](https://github.com/kubernetes-sigs/federation-v2/pull/859/files#diff-66d64392a61d13d392f12c775c53b62fR248)
 - [L367](https://github.com/kubernetes-sigs/federation-v2/pull/859/files#diff-66d64392a61d13d392f12c775c53b62fR367)
 - [L385](https://github.com/kubernetes-sigs/federation-v2/pull/859/files#diff-66d64392a61d13d392f12c775c53b62fR385)
 - [L404](https://github.com/kubernetes-sigs/federation-v2/pull/859/files#diff-66d64392a61d13d392f12c775c53b62fR404)
 - [L422](https://github.com/kubernetes-sigs/federation-v2/pull/859/files#diff-66d64392a61d13d392f12c775c53b62fR422)

Reply:
wrap uses `:` to seperate messages. So, extra `.` is not necessary. I remember related code in controller is already use such kind of convention though a little trivial.

Reply:
Apologies, I may have confused you.  If `forceDeletion == true`, `err` will not be logged at all.  I think the following would be preferable:

```golang
wrappedErr := errors.Wrapf(err, ...)
if !forceDeletion {
   return wrappedErr
}
klog.V(2).Infof("%v", wrappedErr)
```

Reply:
@marun done.